TPLsList.cmake

```diff
@@ -65,6 +65,7 @@ TRIBITS_REPOSITORY_DEFINE_TPLS(
   Cusp            "cmake/TPLs/"    ST
   ROCBLAS         "cmake/TPLs/"    PT
   ROCSPARSE       "cmake/TPLs/"    PT
+  ROCSOLVER       "cmake/TPLs/"    PT
```

Comment:
👍 for adding cmake/TPLs/FindTPLROCSOLVER.cmake , this will allow a follow-up revert of https://github.com/kokkos/kokkos-kernels/pull/2037